spirv_glsl.cpp

```diff
@@ -9606,6 +9606,9 @@ string CompilerGLSL::access_chain_internal(uint32_t base, const uint32_t *indice
 	bool pending_array_enclose = false;
 	bool dimension_flatten = false;
 
+	// If we are translating access to a structured buffer, the first subscript '._m0' must be hidden
+	const bool hide_first_subscript = count > 1 && is_user_type_structured(base) ? 1 : 0;
```

Comment:
Does this handle array of SSBO? I think it's the second index that would have to be skipped in that case.

Reply:
I updated the PR and this works for the following example now as well:
```hlsl
struct Data {
  float4 Color;
};
StructuredBuffer<Data> Colors[2];
float4 PSMain() : SV_Target {
  return Colors[1][3].Color;
}
```